Add <ProductCardOptions /> child component and refine CSS

delawski opened the pull request:
#### Changes proposed in this Pull Request

* Add `ProductCardOptions` sub-component
* Move `PropType` declaration below component definition
* Add tick icon next to the product name if it's purchased
* Add loading state styles, i.e. placeholders for the prices
* Move a SCSS placeholder selectors section to the top of the stylesheet
* Refine `ProductCard`-related styles
* Update devdocs and the readme file

**Mobile**
![Screenshot 2019-10-16 at 21 05 35](https://user-images.githubusercontent.com/478735/66950588-f766ab80-f058-11e9-9ef0-3f239db08601.png)

**Mobile - loading state**
![Screenshot 2019-10-16 at 21 06 02](https://user-images.githubusercontent.com/478735/66950607-00577d00-f059-11e9-9d1f-fda16df4c066.png)

**Desktop**
<img width="702" alt="Screenshot 2019-10-16 at 21 06 31" src="https://user-images.githubusercontent.com/478735/66950632-09484e80-f059-11e9-9f45-75b634c8c7b0.png">

**Desktop - loading state**
![Screenshot 2019-10-16 at 21 06 59](https://user-images.githubusercontent.com/478735/66950643-106f5c80-f059-11e9-8f0c-fd72b08d3cfa.png)

#### Testing instructions

* Go to devdocs and find `ProductCard` UI component
* Confirm that the product options are displayed as in the designs
* Confirm that the product card is responsive and looks as expected on various screen sizes
* Toggle the placeholder state and confirm that the product prices are replaced by a placeholder animation and that this doesn't cause any box dimensions changes (i.e. no unnecessary reflows/repaints)

Fixes n/a

enejb commented on client/components/product-card/README.md:
```diff
@@ -15,20 +15,45 @@ has purchase options, they will be listed below the description.
 See p1HpG7-7nT-p2 for more details.
 
-## Usage
+## <a name="usage"></a>Usage
```
Do we need this?

enejb commented on client/components/product-card/price-group.jsx:
```diff
@@ -23,7 +23,9 @@ const ProductCardPriceGroup = props => {
 			) }
-			<span className="product-card__billing-timeframe">{ billingTimeFrame }</span>
+			<div className="product-card__billing-timeframe">
+				<div className="product-card__billing-timeframe-label">{ billingTimeFrame }</div>
```
What is the reason for having more nesting here?

delawski replied:
Good catch! I needed it at one point but now it's redundant. I'm going to remove it.

enejb commented on client/components/product-card/docs/example.jsx:
```diff
@@ -1,35 +1,32 @@
-				discountedPrice={ 16.99 }
+				isPlaceholder={ isPlaceholder }
+				billingTimeFrame={ isPlaceholder ? null : 'per year' }
+				fullPrice={ isPlaceholder ? null : 25 }
```
Why do we need to change the value based on if it is placeholder?

delawski replied:
It's here to simulate a real scenario when there is no price data fetched yet thus the inner elements of the `ProductCard` are empty and have no dimensions (besides of padding/margin).

enejb replied:
👍

enejb commented:
I think the code looks great!
I did leave some questions in terms of place holder. 

Thanks for adding the placeholder button in the example. It made things easy to test. 
I did notice that there is a bit of a hop when i switch from the placeholder to the regular view. 
I think we should remove it. 

Other then that It looks great!

delawski commented:
@enejb Thank you for your review.

I just tested it out on FF and I can see a hop as well when toggling the loading state. I'm going to work on fixing this.

enejb commented:
Thanks @delawski The issue is now fixed! :shipit:

delawski commented on client/components/product-card/index.jsx:
```diff
@@ -64,20 +53,82 @@ class ProductCard extends Component {
 	}
 }
 
+ProductCard.propTypes = {
```
Brilliant idea! I've addressed it in my latest commit (more about it in a separate, general comment below).

delawski commented:
As discussed, [I've introduced a child component](https://github.com/Automattic/wp-calypso/pull/36834/commits/6cd83bdbbad856a19ab81cc4263ea31ae64d26da) called `<ProductCardOptions />` (note the plural form). It is meant to be passed as a child component to `<ProductCard />` only when needed (i.e. when product options exist).

Since it doesn't make sense to keep the single `<ProductCardOption />` as a separate component (duplication of boilerplate code), I've moved its content to the `<ProductCardOptions />` wrapper.

@tyxla This should be ready for another review.

tyxla commented on client/components/product-card/index.jsx:
```diff
@@ -1,51 +1,47 @@
-		title: PropTypes.oneOfType( [ PropTypes.string, PropTypes.element ] ),
-	};
+const ProductCard = props => {
+	const {
```
Let's list these specifically in the argument, instead of importing the entire `props` and then destructuring it.

delawski replied:
This has been addressed in the [latest commit](https://github.com/Automattic/wp-calypso/pull/36834/commits/c3fb1db60f042e857140c33eb869bc7ac00e84a9).

tyxla commented on client/components/product-card/options.jsx:
```diff
@@ -0,0 +1,71 @@
+import ProductCardPriceGroup from './price-group';
+
+const ProductCardOptions = props => {
+	const {
```
Same as above for moving this one to the arg above so we'd destructure and assign at the same time.

delawski replied:
This has been addressed in the [latest commit](https://github.com/Automattic/wp-calypso/pull/36834/commits/c3fb1db60f042e857140c33eb869bc7ac00e84a9).

delawski commented:
I've addressed the last piece of your feedback, @tyxla. I'm going to merge this as soon as the tests pass.
